my-app/src/components/AddUserPage.tsx

```diff
@@ -1,115 +1,98 @@
 import * as React from 'react';
 import { useState, useEffect } from 'react';
-import { User } from '../types';
-import {validateEmail, validateDate, validateCpf} from './validationHelpers';
+import { User, UserRoleType, UserInputType } from '../types';
+import {validateEmail, validateDate, validateCpf, validatePassword} from './validationHelpers';
+import { createUser } from './UserListQueries';
+import { Redirect } from 'react-router-dom';
 
 
 export const AddUserPage: React.FC = props => {
 
-  const [userName, setUserName] = useState("");
-  const [badName, setBadName] = useState(false);
-  const [userCpf, setUserCpf] = useState("");
-  const [badCpf, setBadCpf] = useState(false);
-  const [userBirthDate, setUserBirthdate] = useState("");
-  const [badDate, setBadDate] = useState(false);
-  const [userEmail, setUserEmail] = useState("");
-  const [badEmail, setBadEmail] = useState(false);
-  const [userRole, setUserRole] = useState("");
-  const [badRole, setBadRole] = useState(false);
+  const [userName, setUserName] = useState({ value:"", valid: true});
+  const [userCpf, setUserCpf] = useState({ value:"", valid: true});
+  const [userBirthDate, setUserBirthdate] = useState({ value:"", valid: true});
+  const [userEmail, setUserEmail] = useState({ value:"", valid: true});
+  const [userRole, setUserRole] = useState({ value:UserRoleType.user, valid: true});
+  const [userPassword, setUserPassword] = useState({ value:"", valid: true});
+  const [redirect, setRedirect] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
 
   function handleChange(event: any) {
     const name = event.target.name;
-
-    switch(name){
-      case "name":
-        setUserName(event.target.value);
-        break;
-      case "cpf":
-        setUserCpf(event.target.value);
-        break;
-      case "birthDate":
-        setUserBirthdate(event.target.value);
-          break;
-      case "email":
-        setUserEmail(event.target.value);
-          break;
-      case "role":
-        setUserRole(event.target.value);
-        break;
-      default:
-    }
+    const value = event.target.value;
+
+    const mapField: { [key: string]: ({value, valid} : {value: any, valid:boolean}) => void } = {
+      name: setUserName,
+      cpf: setUserCpf,
+      birthDate: setUserBirthdate,
+      email: setUserEmail,
+      role: setUserRole,
+      password: setUserPassword,
+    };
+
+    mapField[name]({value: value, valid: true});
   }
 
-  function handleSubmit(){
+  async function handleSubmit(){
 
-    var badEmailtmp = false, badCpftmp = false, badDatetmp = false;
+    var isValid = validateEmail(userEmail.value) && validateDate(userBirthDate.value) && validateCpf(userCpf.value) &&
+    validatePassword(userPassword.value) && !(userName.value == "");
```

Comment:
Cuidado com o operador `==`, às vezes ele não traz exatamente o que queríamos. Dá uma olhada no que os [docs]((https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness)) dizem a respeito

Reply:
Mas nesse caso o "==" já seria o correto (ou indiferente) né? Porque dessa maneira ele checa pelos valores falsy em vez de igualar o tipo também

Reply:
Mas se for pra olhar só os `falsy`, vc pode deixar só `!userName.value` né?